### PR TITLE
Add simple login option for users

### LINKED ISSUE #137: Add simple login option for users

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #137

### CODE REVIEW

File: moviebot/database/db_manager.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            db_name: path to the database file.
+        """
+        self.db_name = db_name
+
+    def __enter__(self):
+        """Returns a cursor to the database."""
+        self.conn = sqlite3.connect(self.db_name)
+        return self.conn.cursor()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
```
Comment (NoB0) [inline moviebot/database/db_manager.py:20]: Nit: missing return type.

File: moviebot/database/db_users.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+class UserDB:
+    def __init__(self, users_db_path: str = "data/users.db") -> None:
+        """Class for managing the users database.
+
+        Args:
+            users_db_path: Path to users database.
+        """
+        self.manager = DatabaseManager(users_db_path)
+
+    def get_user_id(self, username: str) -> int:
```
Comment (NoB0) [inline moviebot/database/db_users.py:19]: The method can return None, so I would update the return type as optional.
```suggestion
    def get_user_id(self, username: str) -> Optional[int]:
```

File: tests/controller/test_server_socket.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+from unittest.mock import MagicMock, patch
```
Comment (NoB0) [inline tests/controller/test_server_socket.py:1]: Nit: add script level docstring.

File: tests/controller/test_server_socket.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+
+@pytest.fixture
+def flask_app() -> Flask:
+    app.testing = True
+    return app
+
+
+@pytest.fixture
+def client(flask_app):
```
Comment (NoB0) [inline tests/controller/test_server_socket.py:16]: Nit: missing return type.

File: tests/controller/test_server_socket.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+
+@pytest.mark.parametrize(
+    "event, expected_event",
+    [
+        ("register", "registration_response"),
+        ("login", "login_response"),
+    ],
+)
+def test_handle_authentication_empty_fields(client, event, expected_event):
```
Comment (NoB0) [inline tests/controller/test_server_socket.py:28]:
```suggestion
def test_handle_authentication_empty_fields(client, event, expected_event) -> None:
```

File: tests/controller/test_server_socket.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    """Test _handle_authentication method for registration with empty fields."""
```
Comment (NoB0) [inline tests/controller/test_server_socket.py:29]: Nit: if I understand correctly it also tests logging in.

File: tests/controller/test_server_socket.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+@pytest.mark.parametrize(
+    "event, expected_event",
+    [
+        ("register", "registration_response"),
+        ("login", "login_response"),
+    ],
+)
+def test_handle_authentication_success(client, event, expected_event):
```
Comment (NoB0) [inline tests/controller/test_server_socket.py:49]:
```suggestion
def test_handle_authentication_success(client, event, expected_event) -> None:
```

**Review by NoB0 — COMMENTED** [review]
Some comments related to the docstrings and one question to address.

File: moviebot/controller/server_socket.py [hunk 8]
```diff
@@ -4,56 +4,145 @@
...
+            emit(
+                event_name,
+                {
+                    "success": False,
+                    "error": "Username and password cannot be empty",
+                },
+            )
+            return
+
+        user_db = UserDB()
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/controller/server_socket.py:51]: Good catch. Made global connection as this is the only class it uses it.

File: moviebot/controller/server_socket.py [hunk 9]
```diff
@@ -4,56 +4,145 @@
...
-    def on_feedback(self, data: dict) -> None:
+    def on_feedback(self, data: Dict[str, Any]) -> None:
         """Receives feedback from client.
 
         Args:
             data: Data received from client.
         """
         logger.info(f"Feedback received: {data}")
+        # user_id = self.active_users.get(request.sid, request.sid)
+        # TODO: Implement feedback logic
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/controller/server_socket.py:125]: Done

File: moviebot/database/db_manager.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+import sqlite3
+
+
+class DatabaseManager:
+    def __init__(self, db_name: str) -> None:
+        """Database manager for providing database context.
+
+        Args:
+            db_name: path to the database file.
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/database/db_manager.py:11]: Done

File: moviebot/database/db_manager.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+        """
+        self.db_name = db_name
+
+    def __enter__(self):
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/database/db_manager.py:15]: Done

File: moviebot/database/db_manager.py (hunk 1 above)
Comment (IKostric) [reply moviebot/database/db_manager.py:20]: Done

File: moviebot/database/db_users.py (hunk 2 above)
Comment (IKostric) [reply moviebot/database/db_users.py:19]: Done

File: tests/controller/test_server_socket.py (hunk 3 above)
Comment (IKostric) [reply tests/controller/test_server_socket.py:1]: Done

File: tests/controller/test_server_socket.py (hunk 4 above)
Comment (IKostric) [reply tests/controller/test_server_socket.py:16]: Done

File: tests/controller/test_server_socket.py (hunk 5 above)
Comment (IKostric) [reply tests/controller/test_server_socket.py:28]: Done

File: tests/controller/test_server_socket.py (hunk 6 above)
Comment (IKostric) [reply tests/controller/test_server_socket.py:29]: Done

File: tests/controller/test_server_socket.py (hunk 7 above)
Comment (IKostric) [reply tests/controller/test_server_socket.py:49]: Done

**Review by NoB0 — APPROVED** [review]
LGTM.

**Pull request merged**